### DISCUSSION TITLE
Is it possible to use @layout with serverside Blazor components?

### DISCUSSION THREAD

**Discussion by FraktalInverTer** [dotnet/aspnetcore#25861] [General] [answered]
I would like to have a cshtml page that loads a serverside blazor component with Html.RenderComponentAsync.

I would like that component to contain another blazor component that is based on a @layout.

Everything works except that the component that has a @layout is just rendered by itself without its layout.

Am I trying to do something unsupported?

**Comment by SteveSandersonMS** [comment, marked as answer]
You can do this but you'll need to use the `<LayoutView>` component (or a `<Router>`), because that's what causes the Blazor component's layout to be rendered.

For example, if your `.cshtml` file renders `MyHost.razor`, containing:

```razor
<LayoutView Layout="@typeof(MainLayout)">
    <SomeOtherComponent />
</LayoutView>
```

... then this will render `SomeOtherComponent` inside the `MainLayout` layout. Using this approach you don't even need to put a `@layout` attribute on your page.

Alternatively inside `MyHost.razor` you could render a Blazor `<Router>` component. This would then take case of picking which other component to render based on the URL and your `@page` attributes, plus it automatically picks whichever layout corresponds to the selected page. For an example of how to use `<Router>`, see the default `App.razor` file inside any newly-created Blazor application.
